concourse/scripts/pipelines-bosh-cloudfoundry.sh

```diff
@@ -18,6 +19,9 @@ cf_grafana_version=$("${SCRIPT_DIR}"/val_from_yaml.rb releases.grafana.version "
 generate_vars_file() {
    cat <<EOF
 ---
+pipeline_name: ${pipeline_name}
+makefile_env_target: ${MAKEFILE_ENV_TARGET:-dev}
+self_update_pipeline: ${SELF_UPDATE_PIPELINE:-false}
```

Comment:
Should this default to `true` and be explicitly disabled for the dev environments in the `Makefile`?

> Self-update will be disabled by default in the dev environments, as we want to be able to apply changes without having to commit and push the code.

This is definitely a use case that we need to allow. Should it be opt-in though?

It's not very clear to newcomers which changes they can make with/without pushing a branch. For people that know what they're doing and want to speed up testing they could do: `SELF_UPDATE_PIPELINE=false make dev`